### PR TITLE
Remove unsafe run sync

### PR DESCRIPTION
**Pull request by yanns**
We should try to use `unsafeRunSync` only at the end.

### CODE REVIEW

File: client/src/test/scala/org/http4s/client/ClientRouteTestBattery.scala [hunk 1]
```diff
@@ -128,29 +128,31 @@ abstract class ClientRouteTestBattery(name: String)
...
-      .through(writeOutputStream[IO](IO.pure(srv.getOutputStream), closeAfterUse = false))
-      .compile
-      .drain
-      .unsafeRunSync()
-  }
+  private def renderResponse(srv: HttpServletResponse, response: IO[Response[IO]]): IO[Unit] =
+    for {
+      resp <- response
+      _ <- IO(srv.setStatus(resp.status.code))
+      _ <- IO(resp.headers.foreach { h =>
```
Comment (isomarcte) [inline client/src/test/scala/org/http4s/client/ClientRouteTestBattery.scala:150]: Nit: Subjectively, I prefer `traverse_` over a `foreach` call.

**Review by isomarcte — APPROVED** [review]

File: client/src/test/scala/org/http4s/client/testroutes/GetRoutes.scala [hunk 2]
```diff
@@ -34,19 +32,18 @@ object GetRoutes {
...
-          .pure[IO],
-        DelayedPath ->
-          F.sleep(1.second) *>
-          Response[IO](Ok).withEntity("delayed path").pure[IO],
-        NoContentPath -> Response[IO](NoContent).pure[IO],
-        NotFoundPath -> Response[IO](NotFound).withEntity("not found").pure[IO],
-        EmptyNotFoundPath -> Response[IO](NotFound).pure[IO],
-        InternalServerErrorPath -> Response[IO](InternalServerError).pure[IO]
-      ))(_.unsafeRunSync())
+  def getPaths(implicit F: Temporal[IO]): Map[String, IO[Response[IO]]] =
```
Comment (rossabaker) [inline client/src/test/scala/org/http4s/client/testroutes/GetRoutes.scala:35]: This one is currently very dumb even on 0.21.

File: client/src/test/scala/org/http4s/client/ClientRouteTestBattery.scala (hunk 1 above)
Comment (rossabaker) [reply client/src/test/scala/org/http4s/client/ClientRouteTestBattery.scala:150]: And wrap the `srv.addHeader` side effect in `IO`?  It's all local, so it doesn't make much difference to me, but that does highlight the effective bit a little better, so, :+1:.

Comment (yanns) [reply client/src/test/scala/org/http4s/client/ClientRouteTestBattery.scala]: https://github.com/http4s/http4s/pull/4096/commits/c26a6cbac341e6721ed6f47c1607e6e7e6d298e3

**Review by rossabaker — APPROVED** [review]

**Pull request merged**